tests/test_cleanup.py

```diff
@@ -176,3 +191,42 @@ def to_monday(d: datetime) -> datetime:
                 (to_monday(base), 90),
             ]
         )
+
+    @pytest.mark.parametrize(
+        "storage_key, create_event_row_for_date", test_data,
+    )
+    @patch("snuba.cleanup.current_time")
+    def test_midnight_error_case(
+        self,
+        current_time: MagicMock,
+        storage_key: StorageKey,
+        create_event_row_for_date: Callable[[datetime, Optional[int]], InsertBatch],
+    ) -> None:
+        def to_monday(d: datetime) -> datetime:
+            rounded = d - timedelta(days=d.weekday())
+            return datetime(rounded.year, rounded.month, rounded.day)
+
+        base = datetime(2022, 1, 29, 0, 4, 37)  # time the script might run
+        current_time.return_value = base
+
+        storage = get_writable_storage(storage_key)
+        clickhouse = storage.get_cluster().get_query_connection(
+            ClickhouseClientSettings.CLEANUP
+        )
+
+        table = storage.get_table_writer().get_schema().get_local_table_name()
+
+        database = storage.get_cluster().get_database()
+
+        parts = cleanup.get_active_partitions(clickhouse, storage, database, table)
+        assert parts == []
+
+        timestamp = datetime(2021, 10, 25)
```

Comment:
What's the scenario that's failing exactly currently? Is it the one where this is a Sunday?

Reply:
Please add the comment. It may also be worth defining the time the script runs as `base = timestamp + timedelta(days=90, minutes=4)` if that is your intention as it's hard to tell the relationship between the 2 dates.